### PR TITLE
deno fixes

### PR DESCRIPTION
**Pull request by crowlKats**
Changes mainly taken from changes i did in #3068

~sadly basically have to duplicate the `wgpu_types::Limits` struct in js; only other solution would be adding serde default attributes to all members of the struct, which doesnt seem too great either...~

Also fixed the default values & order to match the spec & adds serde renames for the texture dimension members.

~If adding serde default attributes would be acceptable, then i would much rather do that than having to need to have similar things spread around~

### CODE REVIEW

**Review by cwfitzgerald — APPROVED** [review]
I don't see a problem with serde defaulting - at least it'll be in the same area of code. Want to get a +1 from a ff person first though before merging.

**Review by nical — APPROVED** [review]
LGTM!

**Pull request merged**
